packages/@react-spectrum/cards/test/CardView.test.js

```diff
@@ -0,0 +1,1139 @@
+    expect(grid).toHaveAttribute('aria-colcount', '1');
+
+    let rowgroups = within(grid).getAllByRole('row');
+    for (let [index, row] of rowgroups.entries()) {
+      expect(row).toHaveAttribute('aria-rowindex', (index + 1).toString());
+      let cell = within(row).getByRole('gridcell');
+      expect(cell).toBeTruthy();
+
+      let image = within(cell).getByRole('img');
+      expect(image).toHaveAttribute('src');
+      expect(within(cell).getByText('Description')).toBeTruthy();
+      expect(within(cell).getByText('PNG')).toBeTruthy();
+      expect(within(cell).getByText('Title', {exact: false})).toBeTruthy();
+
+      if (Name === 'Waterfall layout') {
+        let buttons = within(cell).getAllByRole('button');
+        expect(buttons.length).toEqual(2);
+        expect(buttons[0]).toHaveAttribute('aria-label', 'More actions');
+        expect(buttons[1]).toHaveTextContent('Something');
+      } else {
+        // Grid and Gallery only support quiet cards for now.
+        let actionMenuButton = within(cell).getByRole('button');
+        expect(actionMenuButton).toBeTruthy();
+        expect(actionMenuButton).toHaveAttribute('aria-label', 'More actions');
+      }
+    }
+  });
+
+  describe('Grid layout and some shared gallery layout behavior', function () {
+    it('renders each card with the same height and width', function () {
+      let tree = render(<DynamicCardView />);
+      act(() => {
+        jest.runAllTimers();
+      });
+
+      let grid = tree.getByRole('grid');
+      let wrappers = within(grid).getAllByRole('presentation');
+      let currentTop;
+      let currentLeft;
+      let expectedWidth;
+      let expectedHeight;
+      for (let [index, div] of wrappers.entries()) {
+        if (index === 0) {
+          continue;
+        }
+
+        if (!expectedWidth) {
+          currentTop = div.style.top;
+          currentLeft = div.style.left;
+          expectedWidth = div.style.width;
+          expectedHeight = div.style.height;
+          // default margin size is 24px
+          expect(div.style.top).toEqual('24px');
+          expect(div.style.left).toEqual('24px');
+        } else {
+          expect(div.style.width).toEqual(expectedWidth);
+          expect(div.style.height).toEqual(expectedHeight);
+          if (currentTop === div.style.top) {
+            currentLeft = `${parseInt(currentLeft, 10) + parseInt(expectedWidth, 10) + 24}px`;
+          } else {
+            // default space between the two cards vertically is 48px
+            currentTop = `${parseInt(currentTop, 10) + parseInt(expectedHeight, 10) + 48}px`;
+            currentLeft = '24px';
+            expect(div.style.top).toEqual(currentTop);
+          }
+          expect(div.style.left).toEqual(currentLeft);
+        }
+      }
+    });
+
+    describe('keyboard nav', function () {
+      it('should move focus via Arrow Down', function () {
+        let tree = render(<DynamicCardView />);
+        act(() => {
```

Comment:
what's the initial run timers for?

Reply:
requestAnimationFrame is mocked to be a `jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => setTimeout(cb, 0));` in the beforeAll so need to run timers or the CardView doesn't render with items

Reply:
ah, gotcha, didn't know there was a raf blocking initial render, out of curiosity, where is it? and why don't Table/List tests have this as well?

Reply:
It is in this test's beforeAll section. Table and List mock requestAnimationFrame as `jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb());` aka no timeout